pkg/build/webhook/generic/generic.go

```diff
@@ -43,67 +43,74 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 	}
 
 	if buildCfg.Spec.Source.Git == nil {
-		glog.V(4).Infof("No git source defined for BuildConfig %s/%s, but triggering anyway", buildCfg.Namespace, buildCfg.Name)
-		return revision, envvars, true, err
+		warning := webhook.NewWarning("BuildConfig has no git source defined, ignoring and continuing with build")
+		return revision, envvars, true, warning
 	}
 
 	contentType := req.Header.Get("Content-Type")
 	if len(contentType) != 0 {
 		contentType, _, err = mime.ParseMediaType(contentType)
 		if err != nil {
-			return nil, envvars, false, fmt.Errorf("non-parseable Content-Type %s (%s)", contentType, err)
+			return revision, envvars, false, fmt.Errorf("Error parsing Content-Type: %s", err)
 		}
 	}
 
-	if req.Body != nil && (contentType == "application/json" || contentType == "application/yaml") {
-		body, err := ioutil.ReadAll(req.Body)
-		if err != nil {
-			return nil, envvars, false, err
-		}
+	if req.Body == nil {
+		return revision, envvars, true, nil
+	}
 
-		if len(body) == 0 {
-			return nil, envvars, true, nil
-		}
+	if contentType != "application/json" && contentType != "application/yaml" {
+		warning := webhook.NewWarning("Invalid Content-Type on payload, ignoring payload and continuing with build")
+		return revision, envvars, true, warning
+	}
 
-		var data api.GenericWebHookEvent
-		if contentType == "application/yaml" {
-			body, err = yaml.ToJSON(body)
-			if err != nil {
-				glog.V(4).Infof("Error converting payload to json %v, but continuing with build", err)
-				return nil, envvars, true, nil
-			}
-		}
-		if err = json.Unmarshal(body, &data); err != nil {
-			glog.V(4).Infof("Error unmarshalling payload %v, but continuing with build", err)
-			return nil, envvars, true, nil
-		}
-		if len(data.Env) > 0 && trigger.AllowEnv {
-			envvars = data.Env
-		}
-		if data.Git == nil {
-			glog.V(4).Infof("No git information for the generic webhook found in %s/%s", buildCfg.Namespace, buildCfg.Name)
-			return nil, envvars, true, nil
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return revision, envvars, false, err
+	}
+
+	if len(body) == 0 {
+		return revision, envvars, true, nil
+	}
+
+	var data api.GenericWebHookEvent
+	if contentType == "application/yaml" {
+		body, err = yaml.ToJSON(body)
+		if err != nil {
+			warning := webhook.NewWarning(fmt.Sprintf("Error converting payload to json: %v, ignoring payload and continuing with build", err))
+			return revision, envvars, true, warning
 		}
+	}
+	if err = json.Unmarshal(body, &data); err != nil {
+		warning := webhook.NewWarning(fmt.Sprintf("Error unmarshalling payload: %v, ignoring payload and continuing with build", err))
+		return revision, envvars, true, warning
+	}
+	if len(data.Env) > 0 && trigger.AllowEnv {
+		envvars = data.Env
+	}
+	if data.Git == nil {
+		warning := webhook.NewWarning("No git information found in payload, ignoring and continuing with build")
```

Comment:
i don't think this necessitates a warning.  if we warn here, users who just want to supply env variables are going to get spurious warnings.

I think if

1) we can parse the payload and
2) the result is either git data or env variable data (or both)

we should not warn.